src/main/java/uk/gov/hmcts/reform/unspec/handler/CreateClaimCallbackHandler.java

```diff
@@ -58,52 +73,67 @@ public CreateClaimCallbackHandler(
     }
 
     private CallbackResponse validateClaimValues(CallbackParams callbackParams) {
-        Map<String, Object> data = callbackParams.getRequest().getCaseDetails().getData();
-        ClaimValue claimValue = mapper.convertValue(data.get("claimValue"), ClaimValue.class);
+        CaseData caseData = caseDetailsConverter.to(callbackParams.getRequest().getCaseDetails());
         List<String> errors = new ArrayList<>();
 
+        ClaimValue claimValue = caseData.getClaimValue();
         if (claimValue.hasLargerLowerValue()) {
             errors.add("CONTENT TBC: Higher value must not be lower than the lower value.");
         }
-
+    
+        Map<String, Object> data = callbackParams.getRequest().getCaseDetails().getData();
         if (errors.isEmpty()) {
-            ClaimType claimType = mapper.convertValue(data.get("claimType"), ClaimType.class);
+            ClaimType claimType = caseData.getClaimType();
 
             data.put("allocatedTrack", getAllocatedTrack(claimValue, claimType));
         }
 
         return AboutToStartOrSubmitCallbackResponse.builder()
-                   .data(data)
-                   .errors(errors)
-                   .build();
+            .data(data)
+            .errors(errors)
+            .build();
     }
 
-    private AboutToStartOrSubmitCallbackResponse addIssuedDate(CallbackParams callbackParams) {
-        Map<String, Object> data = callbackParams.getRequest().getCaseDetails().getData();
+    private CallbackResponse issueClaim(CallbackParams callbackParams) {
+        List<String> errors = new ArrayList<>();
+        String authorisation = callbackParams.getParams().get(BEARER_TOKEN).toString();
```

Comment:
This is not needed, we can inject `RequestData` in `SealedClaimFormGenerator` and just call its `authorisation` method there.

Reply:
My two cents, the `RequestData` component seems little dangerous as we are going to need this auth token in some of our async calls as well. My personal preference would be to keep it simple and consistent across the service rather than having two solutions for sync and async methods.

Reply:
Good point, for me using RequestData only for sync methods still makes sense, as it makes the code a little bit simpler - especially if we have to pass auth token between few different methods.